src/jinja2/loaders.py

```diff
@@ -321,8 +321,8 @@ def __init__(
 
         if template_root is None:
             raise ValueError(
-                f"The {package_name!r} package was not installed in a"
-                " way that PackageLoader understands."
+                f"PackageLoader could not find a '{package_path}' directory for the "
+                f"{package_name!r} package."
```

Comment:
I think the original message was about not being able to find the package properly. i.e. At line 315, `roots` would still be an empty list, because it failed to find anything.

So, it seems better to me if the original error is raised if `roots` is empty, otherwise raise this new error message if we don't break from the loop (i.e. add an `else` to the `for` loop).